draft-rha-jose-hpke-encrypt.md

```diff
@@ -157,9 +157,7 @@ This specification describes two modes of use for HPKE in JWE:
   
   *  HPKE Key Encryption mode, where HPKE is used to encrypt a content encryption key (CEK) and the CEK is subsequently used to encrypt the plaintext. This mode supports multiple recipients. This setup is conceptually similar to Key Agreement with Key Wrapping.
 
-When the alg value or enc value is set to any of algorithms registered by this specification then the 'epk' header parameter MUST be present, and it MUST be a JSON Web Key as defined in {{EK}} of this document.
-
-The "ek" member of an 'epk' will contain the base64url encoded "enc" value produced by the encapsulate operation of the HPKE KEM.
+The "ek" member of Protected Header or Unprotected Header will contain the base64url encoded encapsulated key (sometimes called KEM ciphertext) value produced by the encapsulate operation of the HPKE KEM.
```

Comment:
As per the discussion in the mailing list, we wanted to leverage "epk" header parameter that carries keys (see https://mailarchive.ietf.org/arch/msg/jose/zWkQRYP56KBWeMlqiZuzaut8JcY/).

Reply:
Is this is change request? 

Do you mean that you wanted to leveraged the JWK extensions such as `kid` or other properties of the JWK (which allows other JOSN) ?

Reply:
I meant sometime back in the JOSE WG mailing list you had proposed to use "epk" and your reasoning was to keep things as close as possible between key agreement with key wrapping and key encryption (see https://mailarchive.ietf.org/arch/msg/jose/zWkQRYP56KBWeMlqiZuzaut8JcY/).

Reply:
That's true, I've changed my mind, especially after trying to implement ML-KEM HPKE... I think the simpler we can make ek, the better.

Reply:
Yes, ek makes it simpler. We will have to wait for the discussion on whether to use "ek" or "JWE Encrypted Key" is concluded.
Are you referring to https://datatracker.ietf.org/doc/draft-connolly-cfrg-hpke-mlkem/ ?
what challenges did you face when implementing ML-KEM HPKE ?

Reply:
With ML-KEM there is no "enc" that only happens when you get a kemct for HPKE, and it was confusing on how to do the context binding to turn ML-KEM into an HPKE Kem.

Other than that, it was easy and we could have PQ Encryption JWT very soon, if we can get passed how to handle protected headers in new JWE modes.

Reply:
If ML-KEM is used, the KEM context will be "KEM ciphertext" and functions encap/decap will have to change accordingly
but I am not sure of the advantage of using HPKE with ML-KEM. 

The KEM ciphertext does not require to be present in the protected header it could be in the unprotected header, see https://github.com/tireddy2/PQC_JOSE_COSE/blob/main/draft-reddy-cose-jose-pqc-kem.md (KEM ciphertext is bound to the shared secret). 

Why can't ML-KEM be directly used with JWE ?

Reply:
The advantage of using HPKE with ML-KEM is having the same interfaces, and algorithm suits, and being able to move from a dh Kem to a pq kem, without changing other internals.

See also this thread https://mailarchive.ietf.org/arch/msg/cfrg/b8aLlMMbHR-UkxnQgCb4QWi-9b0/

Reply:
You can use KEMs without HPKE.

In the context of JOSE and COSE, why would that be better than using an HPKE KEM?

Making ML-KEM-768+A128KW would be redundant to the HPKE algorithms.

It seems unhelpful to have 2 ways to do basically the same thing.

I think for JOSE and COSE, an ideal outcome would be:

1 way to do KEMs (using HPKE).

Support for PQ, traditional and Hybrids, (using HPKE).

Not have a mixture of PQ, Traditional, Hybrid, HPKE, and regular KEMs.

But it's up to the working groups how complicated they want to make things :)